### PR TITLE
H7 - NUCLEO_H743ZI improvements and bug fixes

### PR DESCRIPTION
**Pull request by rolandvs**
When using different `HSE` value the clock speed can be set in `mpconfigboard.h`, however to match the USB speed a second PLL is used and this was hard coded in `system_stm32.h`.

### CODE REVIEW

File: ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h [hunk 1]
```diff
@@ -58,7 +107,47 @@ void NUCLEO_H743ZI_board_early_init(void);
...
+// SPI5 with NSS select support
+#define MICROPY_HW_SPI5_SCK    (pin_F7)
+#define MICROPY_HW_SPI5_MISO  (pin_F8)
+#define MICROPY_HW_SPI5_MOSI  (pin_F11)
+// or configure no hardware support for NSS and use a different GPIO
+#define MICROPY_HW_SPI5_NSS    (pin_F6)  // optionally
+
+
+// CAN1 bus active on other pins PD0,PD1 and set in can.c
+#define MICROPY_HW_CAN1_RX  (pin_D0)  // still hard coded in can.c
```
Comment (dpgeorge) [inline ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h]: This is no longer hard-coded in can.c

File: ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h [hunk 2]
```diff
@@ -19,26 +26,68 @@ void NUCLEO_H743ZI_board_early_init(void);
...
+
+
+// Available I2C interface on Arduino connector
+#define MICROPY_HW_I2C1_SCL    (pin_B8)  // "D15"
+#define MICROPY_HW_I2C1_SDA    (pin_B9)  // "D14"
+
+#define MICROPY_HW_I2C2_SCL    (pin_F1)
+#define MICROPY_HW_I2C2_SDA    (pin_F0)
+// optional when using SMBUS instead of I2C
+#define MICROPY_HW_I2C2_SMB    (pin_F2)  // not implemented
```
Comment (dpgeorge) [inline ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h]: There's currently no such configuration called I2C_SMB so please remove this line to avoid confusion.

File: ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h [hunk 3]
```diff
@@ -58,7 +107,47 @@ void NUCLEO_H743ZI_board_early_init(void);
 #define MICROPY_HW_USB_VBUS_DETECT_PIN (pin_A9)
 #define MICROPY_HW_USB_OTG_ID_PIN      (pin_A10)
 
-// SD card detect switch
+#define MICROPY_HW_USB_SOF_TP1         (pin_A8) // assigned on this board
+#define MICROPY_HW_USB_FS_POWER        (pin_G6) // if no OTG > LOW!
+#define MICROPY_HW_USB_OVER_CURRENT    (pin_G7) // fault signal
```
Comment (dpgeorge) [inline ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h]: These configs also don't exist... at least not in master.

File: ports/stm32/uart.c [hunk 4]
```diff
@@ -236,6 +246,19 @@ STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
             irqn = UART4_IRQn;
             pins[0] = MICROPY_HW_UART4_TX;
             pins[1] = MICROPY_HW_UART4_RX;
+#if !defined(STM32F4)
```
Comment (dpgeorge) [inline ports/stm32/uart.c]: You shouldn't need to add this guard for STM32F4

File: ports/stm32/uart.c [hunk 5]
```diff
@@ -247,6 +270,18 @@ STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
             irqn = UART5_IRQn;
             pins[0] = MICROPY_HW_UART5_TX;
             pins[1] = MICROPY_HW_UART5_RX;
+#if !defined(STM32F4)
```
Comment (dpgeorge) [inline ports/stm32/uart.c]: Shouldn't need this guard.

File: ports/stm32/uart.c [hunk 6]
```diff
@@ -258,6 +293,7 @@ STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
             irqn = USART6_IRQn;
             pins[0] = MICROPY_HW_UART6_TX;
             pins[1] = MICROPY_HW_UART6_RX;
+#if !defined(STM32F4)
```
Comment (dpgeorge) [inline ports/stm32/uart.c]: Why is this added?

Comment (rolandvs) [reply ports/stm32/uart.c]: When looking at the configurations of all the permissible UARTS on the various `STM32`'s not all ports support hardware handshaking. So for the ports that don't have it, I excluded them in this way, but I imagine that when you leave out the `MICROPY_HW_UARTx_CTS`/`RTS` in the `mpconfigboard.h` it will not give a problem, however It is intended to be used as a guard to not generate code that cannot work with this MCU.

File: ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h (hunk 1 above)
Comment (rolandvs) [reply ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h]: I see this is fixed

File: ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h (hunk 2 above)
Comment (rolandvs) [reply ports/stm32/boards/NUCLEO_H743ZI/mpconfigboard.h]: I will do so

**Comment by dpgeorge** [comment]
@rolandvs are you able to update this PR based on the feedback?  Also if you rebase it on top of the current master code some of the things are now fixed.

**Comment by dpgeorge** [comment]
Closing due to inactivity.  Feel free to reopen with fixes suggested above.
